src/agent/src/rpc.rs

```diff
@@ -1610,30 +1678,50 @@ fn setup_bundle(cid: &str, spec: &mut Spec) -> Result<PathBuf> {
     if spec.root.is_none() {
         return Err(nix::Error::Sys(Errno::EINVAL).into());
     }
-    let spec_root = spec.root.as_ref().unwrap();
+
+    let spec_root = spec
+        .root
+        .as_ref()
+        .ok_or("failed to get spec root as string")
+        .map_err(|e| anyhow!(e))?;
 
     let bundle_path = Path::new(CONTAINER_BASE).join(cid);
     let config_path = bundle_path.join("config.json");
     let rootfs_path = bundle_path.join("rootfs");
 
+    let rootfs_path_name = rootfs_path
+        .to_str()
+        .ok_or("failed to get rootfs path name")
+        .map_err(|e| anyhow!(e))?;
+
     fs::create_dir_all(&rootfs_path)?;
     baremount(
         &spec_root.path,
-        rootfs_path.to_str().unwrap(),
+        rootfs_path_name,
```

Comment:
The better fix would be to make `baremount` take `Path`s rather than bare `&str` making the switches back and forth unnecessary.